File: src/main/java/org/mifos/fineract/services/SelfServiceRegistrationApi.java
package org.mifos.fineract.services;

import retrofit2.http.POST;
import rx.Observable;

public interface SelfServiceRegistrationApi {
    /**
     * @param body (optional)
     * @return Observable&lt;String&gt;
     */
    @POST("self/registration")
    Observable<String> createSelfServiceRegistrationRequest(
            @retrofit2.http.Body String body
    );

    /**
     * @param body (optional)
     * @return Observable&lt;String&gt;
     */
    @POST("self/registration/user")
    Observable<String> createSelfServiceUser(
            @retrofit2.http.Body String body
    );

}

File: src/main/kotlin/org/openapitools/client/apis/SelfServiceRegistrationApi.kt
package org.openapitools.client.apis

import retrofit2.http.Body
import retrofit2.http.POST


interface SelfServiceRegistrationApi {
    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @param body  (optional)
     * @return [kotlin.String]
     */
    @POST("v1/self/registration")
    suspend fun createSelfServiceRegistrationRequest(@Body body: String? = null): String

    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @param body  (optional)
     * @return [kotlin.String]
     */
    @POST("v1/self/registration/user")
    suspend fun createSelfServiceUser(@Body body: String? = null): String

}
